components/hab/src/command/pkg/uninstall.rs

```diff
@@ -0,0 +1,140 @@
+// Copyright (c) 2018 Chef Software Inc. and/or applicable contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::fs;
+use std::path::Path;
+
+use common::package_graph::PackageGraph;
+use common::ui::{Status, UIWriter, UI};
+use error::{Error, Result};
+use hcore::fs as hfs;
+use hcore::package::{all_packages, PackageIdent, PackageInstall};
+
+//use std::iter::IntoIterator;
+
+///
+/// Delete a package and all dependencies which are not used by other packages
+///
+/// 1. We find the fully qualified package ident and all its dependencies
+/// 2. We find all packages on the filesystem and convert them into a graph
+/// 3. We do a BFS on the graph to get the dependencies in order
+/// 4. We check if the specified package has any reverse deps
+///     4a. If there are, we throw an error
+///     4b. If not, we delete the package
+/// 5. For each dependency we check if there are any packages which depend on it
+///     5a. If there are, we skip it
+///     5b. If there are not, we delete it from disk and the graph
+///
+pub fn start(
+    ui: &mut UI,
+    ident: &PackageIdent,
+    fs_root_path: &Path,
+    dryrun: bool,
+    no_deps: bool,
```

Comment:
One thing we've been doing more of lately is trying to use enums rather than bare booleans as arguments, with the idea being that they're more readable, documentable, etc.

So, for example, maybe instead of a boolean `no_deps` we could have something like:
```rust
enum DeletionStrategy {
  PackageOnly
  PackageAndDependencies
}
```
I admit that naming can sometimes be tricky here, but it's proven to be a useful approach in other areas of the codebase, particularly when taking more than one boolean as an argument to a function (then, you get type-safety benefits, too).

Reply:
It makes the code a lot nicer too in terms of reading the logic of the match clauses.  Thanks!